hazelcast/src/main/java/com/hazelcast/instance/Node.java

```diff
@@ -223,6 +225,7 @@ public Node(HazelcastInstanceImpl hazelcastInstance, Config staticConfig, NodeCo
             clusterService = new ClusterServiceImpl(this, localMember);
             textCommandService = new TextCommandServiceImpl(this);
             multicastService = createMulticastService(addressPicker.getBindAddress(), this, config, logger);
+            lockService = new LockServiceImpl(nodeEngine);
```

Comment:
The `LockService` needs the `PartitionService` during construction, so we create it after the `InternalPartitionServiceImpl` has been created (line 224).

During runtime it needs the `OperationService` and `SerializationService`. The `SerializationService` is also created before the `LockService` (line 210). The `OperationServiceImpl` is created in `NodeEngineImpl` line 113, which is done before the `Node` is created.

So all required construction and runtime dependencies of `LockService` should be created before this line.